postgraphiql/src/components/PostGraphiQL.js

```diff
@@ -152,6 +158,32 @@ class PostGraphiQL extends React.PureComponent {
       };
     }
 
+    /**
+     * If we were configured with a graphiqlAuthorizationEventOrigin then we will set up
+     * a message event listener to allow windows with that origin to update our Authorization header.
+     */
+    if (POSTGRAPHILE_CONFIG.graphiqlAuthorizationEventOrigin) {
+      // We want to support a space separated list of origins but avoid a false positive if someone is tricky.
+      const paddedOriginsList = ` ${POSTGRAPHILE_CONFIG.graphiqlAuthorizationEventOrigin} `;
+      console.log('Setting up listener for messages from', paddedOriginsList);
+      window.addEventListener("message", (event) => {
+          if (!paddedOriginsList.includes(event.origin)) {
```

Comment:
This is insecure, I think you mean:

```suggestion
          if (!paddedOriginsList.includes(` ${event.origin} `) || !event.origin) {
```

Though a better approach may be to replace `paddedOriginsList` with

```js
const originsWhitelist = POSTGRAPHILE_CONFIG.graphiqlAuthorizationEventOrigin.split(/\s+/);
```

Reply:
Ah great catch, yep I forgot to pad the testing origin een though I specifically padded the list so it would match when padded!

I was cheating a little here using .includes since I remembered how it worked.  If we have an actual array with split, what is the modern effective way to scan the list for a match?  I could do it in a forEach or something but it feels like there is probably a more modern JS way.

Reply:
You can use `Array.prototype.includes` rather than `String.prototype.includes` 😉 It's available in all current versions of Node (and PostGraphile only supports 8.6+ anyway). Ref: https://node.green/#ES2016-features-Array-prototype-includes